include/olg_scheduler/scheduler.hpp

```diff
@@ -186,52 +188,49 @@ class EventLoop final
     class EventProxy : public Event<time_point>
     {
     public:
-        explicit EventProxy(EventLoop& owner) : Event<time_point>(owner.tree_) {}
         using Event<time_point>::execute;
     };
 
 public:
-    EventLoop()                 = default;
-    EventLoop(const EventLoop&) = delete;
-    EventLoop(EventLoop&&)      = delete;
+    EventLoop() = default;
+
+    EventLoop(const EventLoop&)                = delete;
+    EventLoop(EventLoop&&) noexcept            = delete;
+    EventLoop& operator=(const EventLoop&)     = delete;
+    EventLoop& operator=(EventLoop&&) noexcept = delete;
 
-    ~EventLoop() noexcept
+    ~EventLoop()
     {
         // If this fails, it means that some events have outlived the event loop, which is not permitted.
         assert(isEmpty());
     }
 
-    EventLoop& operator=(const EventLoop&) = delete;
-    EventLoop& operator=(EventLoop&&)      = delete;
-
     /// The provided handler will be invoked with the specified interval starting from (now + period);
     /// if you also need to invoke it immediately, consider using defer().
     /// Returns the event handle on success, empty if out of memory or invalid inputs.
     template <Callback<time_point> Fun>
-    [[nodiscard]] EventPtr<time_point> repeat(const duration period, Fun&& handler)
+    [[nodiscard]] auto repeat(const duration period, Fun&& handler)
     {
         class Impl final : public EventProxy
         {
         public:
-            Impl(EventLoop& owner, const duration per, const Fun& fun) : EventProxy(owner), period_(per), handler_(fun)
+            Impl(EventLoop& owner, const duration per, Fun&& fun) : period_{per}, handler_{std::move(fun)}
             {
-                this->schedule(Clock::now() + period_);
+                this->schedule(Clock::now() + period_, owner.tree_);
             }
-            void execute(const Arg<time_point>& args) final
+
+        private:
```

Comment:
We now expose `Impl` type to user, so we have to make hide private stuff, like `execute`.